src/games/strategy/net/OpenFileUtility.java

```diff
@@ -0,0 +1,54 @@
+package games.strategy.net;
+
+import java.awt.Desktop;
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+
+import javax.swing.JOptionPane;
+
+import games.strategy.debug.ClientLogger;
+
+public class OpenFileUtility {
+
+  /**
+   * Opens a specific file on the user's computer, using whatever default program is used to open such files, using the
+   * local computer's file associations.
+   *
+   * @param file The file to be opened
+   */
+  public static void openFile(final File file) {
+    if (Desktop.isDesktopSupported()) {
+      try {
+        Desktop.getDesktop().open(file);
+      } catch (IOException e) {
+        ClientLogger.logError(e);
+      }
+    } else {
+      logDesktopAPIMessage("File");
```

Comment:
The basic Idea gehind this that if the DesktopAPI is supported, this method should execute without any exceptions - thats why we are logging them - they shouldn't happen.
When java tells us that the DesktopAPI is not supported, then our user has bad luck and has to open the file manually or type in the URL manually - Displaying the Path is a very good suggestion...